script/deploy

```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+set -euo pipefail
+IFS=$'\n\t'
+
+echo "--- Deploying to $HEROKU_APP_NAME"
+if [ "$(git remote | grep $HEROKU_APP_NAME)" = $HEROKU_APP_NAME ]; then
+  git remote remove $HEROKU_APP_NAME
+fi
+git remote add $HEROKU_APP_NAME "https://git.heroku.com/$HEROKU_APP_NAME.git"
+git push --force $HEROKU_APP_NAME "$BUILDKITE_COMMIT":master
```

Comment:
Would it be possible to deploy a particular commit to a particular app if these environment variables somehow made it into `.envrc`? If Buildkite sets up its environment before the pipeline, could they be overridden?

Reply:
Yes. We source from `.envrc` in the `script/ci` script which would override any existing environment variables defined in the pipeline settings.

It's a risk, but I think it's a low risk. Could you help me understand your concern?

Reply:
Thinking more about it, we may not be vulnerable to this. `script/ci` doesn't run `script/deploy`, the pipeline does. Do environment variables carry over between pipeline steps? I don't imagine they would. I imagine they would be run in sub-shells inheriting from the buildkite environment.

Reply:
Ahhh, of course. You are most correct!

We'd have to test it to be sure, but yes, I'd imagine each new step is a new process which means it wouldn't be vulnerable to leaking env vars across steps.

Even if it was we still have a couple of other layers:

1. You can't just git push anywhere. You need valid credentials setup to git push.
2. `master` is a protected branch and there is a review stage.